### PR TITLE
Add a list of test types to the users guide

### PR DESCRIPTION
**Pull request by fischer-ncar**
Add a list of test types to the users guide section 4.

Test suite:
Test baseline:
Test namelist changes:
Test status: [bit for bit, roundoff, climate changing]

Fixes [CIME Github issue #]

User interface changes?:

Update gh-pages html (Y/N)?: Y

Code review:

### CODE REVIEW

**Comment by billsacks** [comment]
Thanks for doing this, @fischer-ncar ! It's not clear to me how these tests are organized. I'd suggest either organizing them alphabetically or by broad category (restart tests, tests of task/thread count, etc.)

**Comment by fischer-ncar** [comment]
@billsacks, I'll arrange them by broad categories (restart tests, task/thread counts, performance, misc.)

**Comment by fischer-ncar** [comment]
I just pushed up a change that has the testtypes arranged with restarts first, then transitioning to thread/mpi tests.  Followed by performance tests, multi-driver/multi-instance tests, misc. "smoke" tests, then data assimilation related tests.

**Comment by billsacks** [comment]
Thanks!

File: doc/source/users_guide/testing.rst [hunk 1]
```diff
@@ -4,9 +4,6 @@
 Testing with create_test
 **************************
 
-.. todo ::
-
-   add list and descriptions on functionality tests (ERS, ERI, ect)
 
 create_test is the tool we use to test cime-driven models. It can be used as a easy way to
```
Comment (bertinia) [inline doc/source/users_guide/testing.rst]: should be "used as an easy"

File: doc/source/users_guide/testing.rst [hunk 2]
```diff
@@ -4,9 +4,6 @@
...
 run a single basic test or can be used to run an entire suite of tests. It has parallelism
```
Comment (bertinia) [inline doc/source/users_guide/testing.rst]: should be "or an entire suite of tests".

File: doc/source/users_guide/testing.rst [hunk 3]
```diff
@@ -25,6 +22,121 @@ the underlying machine and use the default compiler for that machine if this val
...
+
+                         Compare base and seq.
+   DAE                  Data assimilation test, default 1 day, two DA cycles, no data modification.
+   PRE                  Pause-resume test: by default a bit for bit test of pause-resume cycling.
+                         Default 5 hours, five pause/resume cycles, no data modification.
+
+======================= =====================================================================================
+
+
 Each test run by create test will be put through the following mandatory phases:
```
Comment (bertinia) [inline doc/source/users_guide/testing.rst]: This should be: "Each test run by **create_test** includes the following mandatory steps:"

**Comment by rljacob** [comment]
Yes we decided in the Oct 18th meeting to move model-specific usage to an appendix.  And it should say "E3SM" everywhere instead of "ACME".

**Comment by fischer-ncar** [comment]
It looks like section 4.4 Adding tests which is using the update_acme_tests.py should be moved to an appendix too.

**Comment by rljacob** [comment]
That mechanism needs to be generalized so that the list of tests lives in config/acme.   That is on @jgfouca 's todo list.

**Comment by fischer-ncar** [comment]
@bertinia,  I pushed the changes you requested, except for the 4.3 and 4.4 section updates.

**Pull request merged**